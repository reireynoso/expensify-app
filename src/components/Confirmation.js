import React from 'react'

const Confirmation = (props) => {
    return (
        <div className="confirmation-component">    
            <div className="confirmation-component__box">
                <h2 className="confirmation-component__box--title">Are you sure?</h2>
                <p>Removing Expense</p>
                <button className="button" onClick={props.onConfirm}>Cancel</button>
                <button className="button button--remove" onClick={() => this.onRemove()}>Remove</button>
            </div>
        </div>
    )
}

export default Confirmation
