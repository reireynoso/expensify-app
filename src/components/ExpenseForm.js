import React from 'react'

export default class ExpenseForm extends React.Component {
    state = {
        description: "",
        note: "",
        amount: ""

    }

    onDescriptionChange = (e) => {
        const description  = e.target.value;
        this.setState(() => ({
            description
        }))
    }

    onNoteChange = (e) => {
        const note = e.target.value;
        console.log(note)
        this.setState(() => ({
            note
        }))
    }

    amountChange = (e) => {
        const amount = e.target.value;
        if(amount.match(/^\d*(\.\d{0,2})?$/)){
            this.setState(() => ({amount}))
        }
    }
    render(){
        console.log(this.state)
        return (
            <div>
               <form>
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value = {this.state.description}
                        onChange = {this.onDescriptionChange}
                    />

                    <input
                        type="text"
                        placeholder="Amount"
                        onChange={this.amountChange}
                        value={this.state.amount}
                    />
                    
                    <br></br>
                    
                    <textarea
                        placeholder="Add a note for your expense(optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    >
                    
                    </textarea>

                    <button>Add Expense</button>
               </form>
            </div>
        )
    }
}