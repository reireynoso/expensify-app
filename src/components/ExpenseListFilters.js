import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter} from '../actions/filters'
import {sortByAmount, sortByDate, setEndDate, setStartDate} from '../actions/filters'
import {DateRangePicker} from 'react-dates'

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    }

    onDatesChange = ({startDate, endDate}) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    onFocusChange = (calendarFocused) => {
        console.log(calendarFocused)
        this.setState(() => ({
            calendarFocused
        }))
    }
    onSelectChange = (e) => {
        e.target.value === "date" ? this.props.sortByDate() : this.props.sortByAmount()
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    render(){
        return (
        <div className="content-container">
            <div className="input-group">
                <div className="input-group__item">
                    <input 
                        type="text" 
                        className="text-input"
                        onChange={this.onTextChange} 
                        value={this.props.filters.text}
                        placeholder="Search Expenses"
                    />
                </div>
                <div className="input-group__item">
                    <select
                        className="select"
                        value={this.props.filters.sortBy} 
                        onChange={this.onSelectChange}>
                        <option value="date">Date</option>
                        <option value="amount">Amount</option>
                    </select>
                </div>
                <div className="input-group__item">
                    <DateRangePicker
                        startDate={this.props.filters.startDate}
                        endDate={this.props.filters.endDate}
                        onDatesChange={this.onDatesChange}
                        focusedInput={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                </div>
            </div>
        </div>
        )
    }
}

// const ExpenseListFilters = (props) => (
//     <div>
//         <input type="text" onChange={(e) => {
//             props.dispatch(setTextFilter(e.target.value))
//         }} value={props.filters.text}/>
//         <select
//             value={props.filters.sortBy} 
//             onChange={(e) => {
//             e.target.value === "date" ? props.dispatch(sortByDate()) : props.dispatch(sortByAmount())
//         }}>
//             <option value="date">Date</option>
//             <option value="amount">Amount</option>
//         </select>


//     </div>
// )

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount()),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);