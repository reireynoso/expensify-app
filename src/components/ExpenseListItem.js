import React from 'react'
import {connect} from 'react-redux'
import {removeExpense} from '../actions/expenses'

const ExpenseListItem = (props) => {
    const {id, description, amount, createdAt} = props.expense
    return (
    <div>
        <h3>{description}</h3>
        <p>{amount} - {createdAt}</p>
        <button onClick={() => {
            props.dispatch(removeExpense(id))
        }}>Remove</button>
    </div>
    )
}

export default connect()(ExpenseListItem)