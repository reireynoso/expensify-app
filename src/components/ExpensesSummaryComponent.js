import React from 'react'
import {connect} from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import expenseTotal from '../selectors/expenses-total'

export const ExpensesSummaryComponent = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expensesTotal: expenseTotal(selectExpenses(state.expenses, state.filters))
    }
}

export default connect(mapStateToProps)(ExpensesSummaryComponent)