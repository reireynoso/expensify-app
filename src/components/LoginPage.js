import React from 'react'
import {connect} from 'react-redux'
import {startLogin} from '../actions/auth'

export class LoginPage extends React.Component {
    render(){
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify App</h1>
                    <p>Time to get expenses under control.</p>
                    <button onClick={this.props.startLogin}>Login</button>
                </div>
            </div>
        )
    }
} 

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () => dispatch(startLogin())
    }
}

export default connect(null, mapDispatchToProps)(LoginPage)