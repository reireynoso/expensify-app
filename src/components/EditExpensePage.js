import React from 'react'
import {connect} from 'react-redux'
import ExpenseForm from './ExpenseForm'
import {startEditExpense, startRemoveExpense} from '../actions/expenses'
import Confirmation from './Confirmation'

export class EditExpensePage extends React.Component {
    state = {
        confirmation_on: false
    }
    onSubmit = (expense) => {
        // props.dispatch(editExpense(props.expense.id, expense))
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push("/")
    }
    onClick = () => {
        this.setState((prevState) => ({
            confirmation_on: !prevState.confirmation_on
        }))
    }

    onRemove = () => {
        // props.dispatch(startRemoveExpense(props.expense.id))
        this.props.startRemoveExpense(this.props.expense.id)
        this.props.history.push("/")
    }
    render(){
        return(
        <div>
            {
                this.state.confirmation_on && <Confirmation 
                    onCancel={this.onClick} 
                    onConfirm={this.onRemove}
                    optionalText={"Removing Expense"}
                />
            }
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Edit Expense</h1>
                </div>
            </div>
            <div className="content-container">
                <ExpenseForm
                    onSubmit={this.onSubmit}
                    expense={this.props.expense}
                />
                <button className="button button--secondary" onClick={this.onClick}>Remove Expense</button>
            </div>
        </div>
        )
    }
}
// const EditExpensePage = (props) => {
//     // console.log(props)
//     return (
//         <div>
//             <ExpenseForm
//                 onSubmit={(expense) => {
//                     console.log(expense)
//                     props.dispatch(editExpense(props.expense.id, expense))
//                     props.history.push("/")
//                 }}
//                 expense={props.expense}
//             />
//             <button onClick={() => {
//                 props.dispatch(startRemoveExpense(props.expense.id))
//                 props.history.push("/")
//             }}>Remove</button>
//         </div>
//     )
// }

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startEditExpense: (expenseId, expense) => dispatch(startEditExpense(expenseId,expense)),
        startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)