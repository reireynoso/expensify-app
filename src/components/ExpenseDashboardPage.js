import React from 'react'
import ExpensesSummaryComponent from '../components/ExpensesSummaryComponent'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummaryComponent/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
)

export default ExpenseDashboardPage