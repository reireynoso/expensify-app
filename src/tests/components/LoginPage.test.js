import React from 'react'
import {shallow} from 'enzyme'
import {LoginPage} from '../../components/LoginPage'
import toJSON from 'enzyme-to-json'

test("render Login page", () => {
    const wrapper = shallow(<LoginPage/>)
    expect(toJSON(wrapper)).toMatchSnapshot();
})
