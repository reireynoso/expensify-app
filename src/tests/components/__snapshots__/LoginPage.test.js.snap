// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Login page 1`] = `<div />`;
